### PR TITLE
Step four: address core directory files

### PR DESCRIPTION
**Pull request by Monkeychip**
**Step four: tackle core directory files.**

* moved all `core/generic.scss` classes to helper files or to the `utils/element_styling.scss`
* moved all classes from `helper-classes/helpers.scss` that could be moved to more specific helper files (like typography).
* changed `utils/element_styling.scss` to a sass file (was css because it was a direct copy from bulma.css)
* renamed `helper-classes/helpers.scss` to `helper-classes/other.scss` to better follow the organization of helpers stated in the RFC.
* renamed `core/generic.scss` to `core/link.scss` as that was all that was left after I cleaned this up the file.
* add in missing box classes to `core/box.scss`
* remove `core/hero.scss`. The class `hero` or `hero-body` are not used anywhere in the ui.
* add in missing level classes to `core/level.scss`.
* What I thought were only font sizing are used generally throughout the app for margins and padding. Amended comment and started using in the core files when appropriate.
* add in missing menu classes to `core/menu.scss`

If you think screenshots would be valuable for each of these, let me know and I can add them. I was going to but I thought it made it harder to digest. But as a reviewer if you'd prefer to see them, just say the word.

### CODE REVIEW

File: ui/app/styles/core/element-styling.scss [hunk 1]
```diff
@@ -103,32 +103,38 @@ pre {
...
 
 code {
   background-color: transparent;
-  color: #525761;
-  font-size: 0.875em;
+  color: $ui-gray-700;
+  font-size: 1em;
   font-weight: normal;
-  padding: 0.25em 0.5em 0.25em;
+  padding: 0;
```
Comment (Monkeychip) [inline ui/app/styles/core/element-styling.scss:133]: padding: 0 came from the generic file and thus was overriding the bulma elements. I confirmed this by looking at the binary css.

Comment (Monkeychip) [reply ui/app/styles/core/element-styling.scss:133]: same with font size.

File: ui/app/styles/core/link.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  padding: 0;
+  font-weight: $font-weight-semibold;
+  user-select: text;
+
+  &[disabled] {
+    opacity: 0.5;
+    cursor: default;
+  }
+}
+
```
Comment (Monkeychip) [inline ui/app/styles/core/link.scss:24]: ideally we would replace all instance of link-plain with a helper, `is-no-underline`. But I am waiting to do this in a future PR if we determine it's worth it and I have the time.

File: ui/app/styles/components/icon.scss [hunk 3]
```diff
@@ -8,6 +8,7 @@
   display: inline-flex;
   height: 1.5rem;
   justify-content: center;
+  vertical-align: middle;
```
Comment (Monkeychip) [inline ui/app/styles/components/icon.scss:11]: bulma styling for the icon class that I missed in my last PR.

File: ui/app/styles/core.scss
Comment (Monkeychip) [inline ui/app/styles/core.scss:1]: generic --> renamed to link
helpers --> renamed to other

File: ui/app/styles/helper-classes/other.scss
Comment (Monkeychip) [inline ui/app/styles/helper-classes/other.scss:1]: All the helpers removed here I put into the typography helper.

File: ui/app/styles/core/bulma-classes.css [hunk 4]
```diff
@@ -333,10 +333,6 @@ fieldset[disabled] .select select:-ms-input-placeholder,
   }
 }
```
Comment (Monkeychip) [inline ui/app/styles/core/bulma-classes.css:335]: moved to the core/level.scss

File: ui/app/styles/core/level.scss
Comment (Monkeychip) [inline ui/app/styles/core/level.scss:1]: An aside. We override the `core/level.scss` file in several component files. I don't like this, it makes the styling hard to follow. ☹️

**Review by zofskeez — APPROVED** [review]
The _other_ helpers are dwindling! Great job organizing all this 👏
